assemblies/assembly-add-custom-app-file-openshift.adoc

```diff
@@ -0,0 +1,24 @@
+[id='assembly-add-custom-app-file-openshift_{context}']
+= Adding a custom application configuration file to {ocp-brand-name}
+
+To access the {product}, you must add a custom application configuration file to {ocp-brand-name}. In {ocp-short}, you can use the following content as a base template to create a ConfigMap named `app-config-rhdh`:
```

Comment:
Quick question about the template for the custom config file. For the operator, we recommended filling in additional fields like the base URL and CORS origin URL. 
So to me, the template would be slightly different for the operator (and a secret would need to be created to store the `BACKEND_SECRET`). Otherwise, the application will not start with the Operator (at least in 1.1), because of a missing `backend.auth.keys.secret`.

See https://access.redhat.com/documentation/en-us/red_hat_developer_hub/1.1/html/administration_guide_for_red_hat_developer_hub/assembly-install-rhdh-ocp#proc-add-custom-app-config-file-ocp_admin-rhdh

```yaml
kind: ConfigMap
apiVersion: v1
metadata:
  name: app-config-rhdh
data:
  "app-config-rhdh.yaml": |
    app:
      title: Red Hat Developer Hub
      baseUrl: https://backstage-<CUSTOM_RESOURCE_NAME>-<NAMESPACE_NAME>.<OPENSHIFT_INGRESS_DOMAIN>
    backend:
      auth:
        keys:
          - secret: "${BACKEND_SECRET}"
      baseUrl: https://backstage-<CUSTOM_RESOURCE_NAME>-<NAMESPACE_NAME>.<OPENSHIFT_INGRESS_DOMAIN>
      cors:
        origin: https://backstage-<CUSTOM_RESOURCE_NAME>-<NAMESPACE_NAME>.<OPENSHIFT_INGRESS_DOMAIN>
```

Reply:
The template in this PR is okay for Helm, as Helm already generates a default app-config file (+ a random backend auth secret), where the base URL and CORS origin URL are constructed from the `global.clusterRouterBase` parameter.

(But I think we can use the template above for both Helm and the Operator).

Reply:
That should be added to the operator as part of the feature parity between helm <-> operator

Reply:
Sure, let's capture this in the feature parity epic ([RHIDP-2311](https://issues.redhat.com/browse/RHIDP-2311)). But until this is implemented, I think our product docs should reflect the current state.

Reply:
@rm3l is the procedure you linked specific to adding a custom app config file to OCP **using the Operator**? 

If _no,_ then I'll add only the parts of the linked procedure that are applicable to this one. 

If _yes_, then I will take a different approach (like edit the procedure you linked to specify that and then link to it from this doc that we are working on). Ultimately, I see a need to consolidate all of the custom app config file content into one place, rather than leaving it scattered between both the Admin and Getting Started guides, but will address that issue in a future Jira/PR.

Reply:
Note: The linked content is currently under a Helm chart chapter in the Admin guide, but seems that it may not actually be specific to the Helm chart, but rather general or specific to the Operator only.

Reply:
> @rm3l is the procedure you linked specific to adding a custom app config file to OCP **using the Operator**?
> 
> If _no,_ then I'll add only the parts of the linked procedure that are applicable to this one.
> 
> If _yes_, then I will take a different approach (like edit the procedure you linked to specify that it is for the Operator and then link to it from this doc that we are working on).

I would say _yes_, this is specific to the operator at this time. The URLs mentioned in the custom app-config need to be provided explicitly by the user when using the operator, otherwise some parts of the application (like GitHub auth) might not work properly.
For Helm, adding those URLs is not needed explicitly in the custom app-config, as long as users set the correct router hostname (`global.clusterRouterBase`) in their `values` file, as depicted in https://access.redhat.com/documentation/en-us/red_hat_developer_hub/1.1/html/administration_guide_for_red_hat_developer_hub/assembly-install-rhdh-ocp#proc-install-rhdh-ocp-helm_admin-rhdh

> Ultimately, I see a need to consolidate all of the custom app config file content into one place, rather than leaving it scattered between both the Admin and Getting Started guides, but will address that issue in a future Jira/PR.

+1. Definitely agree to have all this consolidated into one place.